src/wallet/rpcdump.cpp

```diff
@@ -131,7 +131,11 @@ UniValue importprivkey(const JSONRPCRequest& request)
     CKeyID vchAddress = pubkey.GetID();
     {
         pwallet->MarkDirty();
-        pwallet->SetAddressBook(vchAddress, strLabel, "receive");
+
+        // We don't know which corresponding address will be used; label them all
+        for (const auto& dest : GetAllDestinationsForKey(pubkey)) {
```

Comment:
If I'm reading this correctly, if you import with importpubkey you can downgrade and you'll still get the segwit watchonly, but not if you do an importprivkey? This seems strange, but either way the behavior should be documented in the help text of both importprivkey and importpubkey at least until 0.17 (or in release notes).

Reply:
I don't understand.

Reply:
https://github.com/bitcoin/bitcoin/pull/11403#discussion_r157384169

> If I'm reading this correctly, if you import with importpubkey you can downgrade and you'll still get the segwit watchonly, but not if you do an importprivkey?

@TheBlueMatt can you describe what the documentation should say about this?

Reply:
I discussed this IRL with @TheBlueMatt; what I believe he meant is that importing private keys does not explicitly add the related scripts for those keys to the wallet (possibly hurting the ability to detect outputs to such keys after downgrade). Fixed.